### PR TITLE
#186 Workaround to find more than 1000 repositories

### LINKED ISSUE #186: impossible to find more than 1000 repositories
At the moment, there is a limitation in GitHub API: "search" can only retrieve up to 1000 repositories. Let's find a workaround, so that we can find more than 1000. See https://stackoverflow.com/questions/37602893/github-search-limit-results

### PR DESCRIPTION
**Pull request by lueFlake**
resolves #186

### CODE REVIEW

File: steps/discover-repos.rb [hunk 1]
```diff
@@ -28,18 +28,18 @@
...
-max = 1000
+require 'date'
 
 opts = Slop.parse do |o|
   o.string '--token', 'GitHub access token', default: ''
   o.boolean '--dry', 'Make no round-trips to GitHub API (for testing)', default: false
   o.integer '--total', 'Total number of repos to take from GitHub', required: true
   o.integer '--pause', 'How many seconds to sleep between API calls', default: 10
   o.integer '--page-size', 'Number of repos to fetch in one API call', default: 100
-  o.integer '--min-stars', 'Minimum GitHub stars in each repo', default: max
```
Comment (yegor256) [inline steps/discover-repos.rb:40]: @lueFlake what is the point of this change?

Comment (lueFlake) [reply steps/discover-repos.rb:40]: @yegor256 why would we need to keep "max" variable? It is used only once here and i dont understand logic begind it in this line

**Comment by yegor256** [comment]
@lueFlake looks good, but there are some failures in the build

**Comment by lueFlake** [comment]
@yegor256 yep, i did something wrong while refactoring, now fixed

**Comment by lueFlake** [comment]
@yegor256 fixed issues with build, but it fails on `test-report.sh` on macos, i dont have an access for macos, so i dont understand how shoud i fix it

File: steps/discover-repos.rb [hunk 2]
```diff
@@ -97,44 +82,110 @@ def mock_reps(page, size, licenses)
...
 end
 
-def cooldown(opts, found)
-  puts "Let's sleep for #{opts[:pause]} seconds to cool off GitHub API \
-(already found #{found.count} repos, need #{opts[:total]})..."
-  sleep opts[:pause]
+def process_year(year, github, context)
+  query = build_query(year, context[:opts])
+  puts "Querying for repositories created in #{year}..."
+
```
Comment (yegor256) [inline steps/discover-repos.rb:88]: @lueFlake what's the point of this empty line? https://www.yegor256.com/2014/11/03/empty-line-code-smell.html

File: steps/discover-repos.rb [hunk 3]
```diff
@@ -97,44 +82,110 @@ def mock_reps(page, size, licenses)
...
+    'mirror:false',
+    'archived:false', 'template:false', 'NOT', 'android'
+  ].join(' ')
+end
+
+def loop_through_pages(query, github, context)
+  page = 0
+  loop do
+    break if context[:found].count >= context[:opts][:total]
+
```
Comment (yegor256) [inline steps/discover-repos.rb:109]: @lueFlake better avoid them (empty lines)

**Comment by yegor256** [comment]
@lueFlake macOS build is broken in master :( https://github.com/yegor256/cam/issues/381

File: steps/discover-repos.rb (hunk 3 above)
Comment (lueFlake) [reply steps/discover-repos.rb]: @yegor256 removed blank lines, forgot to remove them after debug, my bad

**Comment by yegor256** [comment]
@lueFlake thanks!

**Pull request merged**